_posts/2022-09-03-aphp-user-story.adoc

```diff
@@ -0,0 +1,74 @@
+
+For that last part, our choices leaned toward Wildfly/Jboss. We chose to package our applications in a single deliverable containing front end and back end due to the small size of theres and to facilitate deployment. Our front end is rarely evolving, or to be more exact it evolves only when the back end is doing so. This allows our teams to quickly deploy new features. The size of our Information Network (hundreds of major apps and thousands of working instances) makes it so its global evolution is slow paced.
+
+Using containers made us revise our choices. A key point in our decision was to keep working with technologies already mastered by our teams, and thus ensuring that any deployment or development of an app would’t imply a total start from scratch, but rather the use of preexisting mods. We wanted an evolution and not a revolution.
+
+== Quarkus
+
+The DSN/DEV kept close attention to Quarkus while researching other technologies aswell. With the release 1.x of Quarkus, it became evident that it would become the tool that would take us in the right direction. 
+
+To manage the data flux between our apps, a flux supervisor tool was created, long before we chose to work with JEE. This tool was based on servlet, and over its 10 years of existence a heavy stratification took place. As it’s still performing properly, this app was chosen as our Quarkus test subject. We could try to develop a replacement and if we encountered any unexpected obstacle we could go back to the existing app.
+
+Our wish was to capitalize on recurring elements. From experience, our teams are able to know what can be reused and where. For a long time, we developed in a modular fashion. Each project being an assembly of mods. We are used to externalizing these mods to independent projects.
+
+We rapidly managed to produce a prototype. Then a partner school was tasked to translate the two basic functions of this app from servlet to JEE/Angular. Once it was done, we needed to go from JEE/Angular to Quarkus/Angular. Which took only one week for one person, outside of their regular duties. Quarkus was indeed the perfect tool for the job.
+
+Based on this experience, the project started. The supervising app was nothing more than a big mod cluster. Each mods providing its own function, it became simple to add any needed function by adding a new mod.
+
+We were forced to overcome some limitations such as Quarkus Panache preventing us from correctly accessing some of our Oracle database. But we manage quite easily to follow Quarkus evolutions and re adapt our code to take these evolution into account.
+
+Regarding  Quarkus-Quinoa, which resembles our Angular approach. Our 10-year-old packaging still performing perfectly and being totally automatized, made it so we chose to skip Quinoa.
```

Comment:
```suggestion
Regarding Quarkus-Quinoa, which resembles our Angular approach: Because our 10-year-old packaging is still performing perfectly and totally automatized, we chose to not use Quinoa.
```